### DISCUSSION TITLE
spatie/pdf not work on production

### DISCUSSION THREAD

**Discussion by Darnes1958** [spatie/laravel-pdf#204] [Q&A]
I install spatie/pdf and puppeteer , it work well in local but when deployment to server it is not work.
fisrt error about chaching
as I search I had create .puppeteerrc.cjs and add this content : 
const {join} = require('path');

/**
 * @type {import("puppeteer").Configuration}
 */
module.exports = {
    // Changes the cache location for Puppeteer.
    cacheDirectory: join(__dirname, '.cache', 'puppeteer'),
};

the error is disapeare but another error is rise after take log period of process: 
Page.printToPDF timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed. at <instance_members_initializer>

the server is windows serve with Plesk panel

**Comment by shelleychristie** [comment]
I had the exact same issue. Worked on local, not in production. After more than a day of scouring and trying to fix the error, I found this [article](https://fly.io/laravel-bytes/invoice-pdf-generation-with-browsershot/) which mentioned using noSandbox() on browsershot to avoid errors. This is recommended only if you absolutely trust the source or made it yourself. Here is a snippet from my controller, it now works in prod.

```
use Spatie\Browsershot\Browsershot;

return Pdf::view('invoices.pdf', compact('order'))
            ->withBrowsershot(function (Browsershot $browsershot) {
                // prevents puppeteer errors with navigation
                $browsershot->noSandbox();
            })
            ->format('a4')
            ->landscape()
            ->name($filename);
```
